### PR TITLE
Add `format` in path params

### LINKED ISSUE #3070: [protoc-gen-openapiv2] `format` not being generated for path parameters
## 🐛 Bug Report

The `format` field in the openapi schema object is not being populated for path parameters even if specified in the `openapiv2_operation`.

For instance:

```proto
message FormatOutputNOK {
    string my_field_nok = 1 [
      (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
        title: "Credential identifier"
        description: "My field, the format is NOT generated"
        format: "uuid"  // <----- this field is not being generated in the OpenAPI spec.
    }];
}
```

````yaml
parameters:
    - name: myFieldNok
      description: My field, the format is NOT generated
      in: path
      required: true
      type: string
      # <---- there is no "format: uuid" being added here
````

## To Reproduce

1. Generate the OAS file with the configuration below.
2. Notice there is no `format` being generated

<details>
  <summary>buf.gen.yaml</summary>

  ```proto
version: v1
managed:
  enabled: true
  go_package_prefix:
    default: demo/demo/demo
plugins:
  - name: openapiv2
    out: .
    strategy: all
    opt:
      - allow_merge=true
      - merge_file_name=registry-manager
      - allow_repeated_fields_in_body=true
      - omit_enum_default_value=true
      - disable_service_tags=true
      - output_format=yaml
  ```
</details>

<details>
  <summary>Simple proto file</summary>

  ```proto

syntax = "proto3";

import "google/api/field_behavior.proto";
import "google/protobuf/timestamp.proto";
import "protoc-gen-openapiv2/options/annotations.proto";
import "google/api/annotations.proto";
import "google/protobuf/field_mask.proto";

package demo.v1;

service MyService {
    rpc MyOperation(FormatOutputNOK) returns (FormatOutputOK) {
      option (google.api.http) = {
        delete: "/api/v1/myservice/{my_field_nok}"
      };

      option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
        summary: "My service";
        description: "My service description";
      };
    }
}

message FormatOutputNOK {
    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
      json_schema: {
        title: "My operation request"
        description: "My operation request description"
      }
    };

    string my_field_nok = 1 [
      (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
        title: "Credential identifier"
        description: "My field, the format is NOT generated"
        format: "uuid"
    }];
}

message FormatOutputOK {
    string my_field_ok = 1 [
     (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
        title: "My field OK"
        description: "My field, the format is generated"
        format: "uuid"
    }];
}

  ```
</details>

## Expected behavior

````yaml
....
      type: string
      format:  uuid # <--- this should be generated
````

## Actual Behavior

````yaml
....
      type: string
      <nothing is generated>
````

## Your Environment

buf 1.10.0

## Extra

I haven't gone through the code thoroughly, but at a glance, I'd say the `format` is being set to `""` for path params here, even if specified in the``openapiv2_field` option.

See
https://github.com/grpc-ecosystem/grpc-gateway/blob/0e7b2ebe117212ae651f0370d2753f237799afdf/protoc-gen-openapiv2/internal/genopenapi/template.go#L1109

### PR DESCRIPTION
**Pull request by antgamdia**
#### References to other Issues or PRs

Fixes #3070

#### Have you read the [Contributing Guidelines](https://github.com/grpc-ecosystem/grpc-gateway/blob/main/CONTRIBUTING.md)?

Yep!

#### Brief description of what is fixed or changed

This PR is adding the `format` field in the OpenAPI generator, which was wrongly set to `""` in some cases.

#### Other comments

I haven't been able to add a test case for the actual failing scenario (a path param with a `format` in the field options). However, I have added some test cases covering the "string with format" scenario in other params (headers and query).
If you can point me to the proper test function, I'll happily add a test case.

### CODE REVIEW

File: protoc-gen-openapiv2/internal/genopenapi/template.go [hunk 1]
```diff
@@ -793,7 +793,7 @@ func primitiveSchema(t descriptorpb.FieldDescriptorProto_Type) (ftype, format st
 		// NOTE: in OpenAPI specification, format should be empty on boolean type
 		return "boolean", "", true
 	case descriptorpb.FieldDescriptorProto_TYPE_STRING:
-		// NOTE: in OpenAPI specification, format should be empty on string type
+		// NOTE: in OpenAPI specification, can be empty on string type
```
Comment (johanbrandhorst) [inline protoc-gen-openapiv2/internal/genopenapi/template.go:796]: While we're here, can we add a reference to the data types part of the spec to the docstring of this function? This link: https://swagger.io/specification/v2/#data-types.

File: protoc-gen-openapiv2/internal/genopenapi/template.go [hunk 2]
```diff
@@ -1133,6 +1133,9 @@ func renderServices(services []*descriptor.Service, paths openapiPathsObject, re
 						desc = schema.Description
 						defaultValue = schema.Default
 						extensions = schema.extensions
+						if paramFormat == "" {
+							paramFormat = schema.Format
```
Comment (johanbrandhorst) [inline protoc-gen-openapiv2/internal/genopenapi/template.go:1137]: Please add a comment here to describe why this is happening
```suggestion
							// If the is no mandatory format based on the field,
							// allow it to be overridden by the user.
							paramFormat = schema.Format
```

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]
This looks like a great start, thank you!

**Comment by antgamdia** [comment]
> This looks like a great start, thank you!

Thank you! I've added the current suggestions you made, however, I have to wait a few days (perhaps more in these Christmas days?) til I get the OK for signing the CLA in my company. As soon as I have it, I'll sign it straight away :)

**Comment by antgamdia** [comment]
> however, I have to wait a few days (perhaps more in these Christmas days?) til I get the OK for signing the CLA in my company. As soon as I have it, I'll sign it straight away :)

Done :)

**Review by johanbrandhorst — APPROVED** [review]

**Comment by johanbrandhorst** [comment]
Looks like some files need to be regenerated after this change, see CONTRIBUTING.md. Thanks!

**Comment by antgamdia** [comment]
> Looks like some files need to be regenerated after this change, see CONTRIBUTING.md. Thanks!

Yep, sorry, you're right, I missed an autogenerated file :sweat_smile: Adding it now!

**Comment by johanbrandhorst** [comment]
Thanks for your contribution!

**Pull request merged**